financial-connections/src/main/java/com/stripe/android/financialconnections/navigation/Destination.kt

```diff
@@ -232,37 +249,40 @@ internal sealed class Destination(
             ?.let { value -> Pane.entries.firstOrNull { it.value == value } }
 
         const val KEY_REFERRER = "referrer"
+        const val KEY_NEXT_PANE_ON_DISABLE_NETWORKING = "next_pane_on_disable_networking"
     }
 }
 
-internal fun String.appendParamValues(params: List<String>): String {
-    return buildString {
-        append(this@appendParamValues)
-        params.forEach { append("/$it") }
+internal fun String.appendParamValues(params: Map<String, String?>) = buildString {
+    append(this@appendParamValues)
+    if (params.isNotEmpty()) {
+        append("?")
+        params.entries
+            .filter { it.value != null }
+            .joinToString("&") { (key, value) -> "$key=$value" }
+            .let { append(it) }
```

Comment:
we were using path params instead of query params, so that "nullable params" where not supported. Refactored so that we build destination urls with query params.